src/components/seo.tsx

```diff
@@ -5,12 +5,16 @@
  * See: https://www.gatsbyjs.org/docs/use-static-query/
  */
 
-import React from 'react'
+import React, { ReactElement } from 'react'
 import Helmet from 'react-helmet'
 import { useStaticQuery, graphql } from 'gatsby'
 
 interface PureSEOProps {
-  data: any
+  data: {
+    site: {
+      siteMetadata: { description: string; title: string; author: string }
```

Comment:
Super, že je přidané otypování. Zvážil bych přesun do vlastního interfacu, klidně jen lokálně.

Reply:
Jasně, dává smysl. Změnil jsem na: 

```typescript
interface SiteMetadata {
  description: string
  title: string
  author: string
}

type GraphQLData = {
  site: {
    siteMetadata: SiteMetadata
  }
}

interface PureSEOProps {
  data: GraphQLData
  description?: string
  lang?: string
  meta?: []
  title?: string
}
```

To zanoření `site.siteMetadata`, ale zatím mi tady přijde zbytečné vytahovat do jednoho typu navíc.